packages/builder/src/builderStore/store/frontend.js

```diff
@@ -481,6 +481,57 @@ export const getFrontendStore = () => {
           return null
         }
 
+        let dataSourceField = definition.settings.find(
```

Comment:
For this first half - we actually have a util called `getComponentSettings` in `componentUtils.js` which will get the full list of settings for  component type, including flattening and merging settings inside section. We could use that here to reduce a lot of this code!